uiowa/uiowa
# Feature: banner button options

Resolves https://github.com/uiowa/uiowa/issues/5086

Todo:
- [x] add "button font" group with a "Thin" option to set alternate button style
- [x] update button styles to just read as "gold" , "black", "white", and "outline"
- [x] Review and merge https://github.com/uiowa/uids/pull/681 and attach as a UIDS release to this pull request
- [x] Update hook for banners setting default button style to `primary`
- [x] ~Add these button options to CTAs since these button style updates from UIDS will effect these buttons: https://github.com/uiowa/uiowa/issues/5149~ Added temporary fix: https://github.com/uiowa/uiowa/pull/5146/commits/128b884d65eaf46e3ddbe48bb4f74b0cf835c034

# How to test

- `blt frontend`
- `ddev blt ds --site=sandbox.uiowa.edu`
- Test on a number of banner examples:
- - https://sandbox.prod.drupal.uiowa.edu/banners-imagesvideos-w-black-and-white-gradients
- - https://sandbox.prod.drupal.uiowa.edu/banner-backgrounds-w-gradient-dark
- - https://sandbox.prod.drupal.uiowa.edu/banner-backgrounds-w-gradient-light
- - https://sandbox.prod.drupal.uiowa.edu/banners-and-buttons
- Verify that update hook set existing banners with gold button style
- Verify that newly created banners are set with the gold button style
- Verify that you can adjust and change the button styles on a banner using one link and multiple links. 
- Verify that you can use one of the outline styles against the dark gradient and that the outline turns white and the text turns white.  Test this option using the banner stacked style as well. 
- test all button color options with the button font "thin" option.
